webapp/src/components/cardDetail/cardDetailContents.test.tsx

```diff
@@ -0,0 +1,258 @@
+                {
+                    color: 'propColorDefault',
+                    id: 'property_value_id_1',
+                    value: 'Jean-Luc Picard',
+                },
+                {
+                    color: 'propColorDefault',
+                    id: 'property_value_id_2',
+                    value: 'William Riker',
+                },
+                {
+                    color: 'propColorDefault',
+                    id: 'property_value_id_3',
+                    value: 'Deanna Troi',
+                },
+            ],
+        },
+    ]
+
+    const card = TestBlockFactory.createCard(board)
+
+    test('should match snapshot', async () => {
+        const component = wrapProviders((
+            <CardDetailContents
+                id='test-id'
+                card={card}
+                contents={[]}
+                readonly={false}
+            />
+        ))
+
+        let container: Element | undefined
+        await act(async () => {
+            const result = render(component)
+            container = result.container
+        })
+        expect(container).toMatchSnapshot()
+    })
+
+    test('should match snapshot with contents array', async () => {
+        const contents = [TestBlockFactory.createDivider(card)]
+        const component = wrapProviders((
+            <CardDetailContents
+                id='test-id'
+                card={card}
+                contents={contents}
+                readonly={false}
+            />
+        ))
+
+        let container: Element | undefined
+        await act(async () => {
+            const result = render(component)
+            container = result.container
+        })
+        expect(container).toMatchSnapshot()
+    })
+
+    test('should match snapshot after onBlur triggers', async () => {
+        const component = wrapProviders((
+            <CardDetailContents
+                id='test-id'
+                card={card}
+                contents={[]}
+                readonly={false}
+            />
+        ))
+
+        let container: Element | undefined
+        await act(async () => {
+            const result = render(component)
+            container = result.container
+        })
+        const markdownEditorField = container!.querySelectorAll('.octo-editor-preview.octo-placeholder')
+        expect(markdownEditorField).toBeDefined()
+        fireEvent.click(markdownEditorField[markdownEditorField.length - 1]!)
+
+        const onFocusEvent = new FocusEvent('focus', {
+            view: window,
+            bubbles: true,
+            cancelable: true,
+        })
+
+        const onBlurEvent = new FocusEvent('blur', {
+            view: window,
+            bubbles: true,
+            cancelable: true,
+        })
+
+        const textareaContainer = container!.querySelectorAll('.CodeMirror.cm-s-easymde.CodeMirror-wrap')
+        const textarea = textareaContainer[textareaContainer.length - 1].querySelector('textarea')
```

Comment:
Same question about `textareaContainer`: do we expect to have single element in this NodeList?

Reply:
No we don't. We tried `querySelector` but we were getting the wrong element. So we had to use All and the last one happens to be the one I want via trial and error through an actual browser.